source/postprocess/heat_flux_map.cc

```diff
@@ -0,0 +1,221 @@
+                                        update_q_points       | update_JxW_values);
+
+      typename MaterialModel::Interface<dim>::MaterialModelInputs in(fe_face_values.n_quadrature_points, this->n_compositional_fields());
+      typename MaterialModel::Interface<dim>::MaterialModelOutputs out(fe_face_values.n_quadrature_points, this->n_compositional_fields());
+
+      std::vector<Tensor<1,dim> > temperature_gradients (quadrature_formula.size());
+      std::vector<std::vector<double> > composition_values (this->n_compositional_fields(),std::vector<double> (quadrature_formula.size()));
+
+      // have a stream into which we write the data. the text stream is then
+      // later sent to processor 0
+      std::ostringstream output;
+      std::vector<std::pair<Point<dim>,double> > stored_values;
+
+      // loop over all of the surface cells and evaluate the heatflux
+      typename DoFHandler<dim>::active_cell_iterator
+      cell = this->get_dof_handler().begin_active(),
+      endc = this->get_dof_handler().end();
+
+      for (; cell!=endc; ++cell)
+        if (cell->is_locally_owned())
+          for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+            if (cell->at_boundary(f))
+              {
+                fe_face_values.reinit (cell, f);
+
+                //evaluate position of heatflow to write into output file
+                const Point<dim> midpoint_at_surface = cell->face(f)->center();
+
+                // get the various components of the solution, then
+                // evaluate the material properties there
+                fe_face_values[this->introspection().extractors.temperature].get_function_gradients (this->get_solution(),
+                    temperature_gradients);
+
+                for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                  fe_face_values[this->introspection().extractors.compositional_fields[c]].get_function_values(this->get_solution(), composition_values[c]);
+
+
+                for (unsigned int i=0; i<fe_face_values.n_quadrature_points; ++i)
+                  {
+                    for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                      in.composition[i][c] = composition_values[c][i];
+                  }
+
+                this->get_material_model().evaluate(in, out);
+
+
+                // Calculate the normal conductive heat flux given by the formula
+                //   j = - k * n . grad T
+
+                double normal_flux = 0;
+                for (unsigned int q=0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    const double thermal_conductivity
+                      = out.thermal_conductivities[q];
+
+                    normal_flux += -thermal_conductivity *
+                                   (temperature_gradients[q] * fe_face_values.normal_vector(q)) *
+                                   fe_face_values.JxW(q);
```

Comment:
so what you compute here is actually the *integrated* output flux, integrated over each face. This really *is* the definition of a flux (in Watts), but I suspect that you really want instead a *flux density* in W/m^2. To do this, after integrating it up over the face, you'll have to divide it by the area of the face. The area is simply the sum of all of the JxW values.